### ISSUE TITLE
Non-zero wait status: 11 in testt

### ISSUE THREAD

**Issue by robertlemmen** [rakudo/rakudo#2567]
builds of 2018.12 on ppc64el failed during tests:

https://buildd.debian.org/status/fetch.php?pkg=rakudo&arch=ppc64el&ver=2018.12-1&stamp=1545495855&raw=0

the problem seems to happen only rarely, but running t/02-rakudo/08-inline-native-arith.t in a loop eventually led to this slightly information-free reproduction:

```
Test Summary Report
-------------------
t/08-performance/99-misc.t (Wstat: 11 Tests: 0 Failed: 0)
  Non-zero wait status: 11
  Parse errors: No plan found in TAP output
Files=1, Tests=0,  1 wallclock secs ( 0.02 usr  0.01 sys +  0.56 cusr  0.02 csys =  0.61 CPU)
Result: FAIL

```

Please let me know what I can do to help debug this

**Comment by robertlemmen** [comment]
this may actually be a duplicate of #2564 ...

**Comment by AlexDaniel** [comment]
@robertlemmen #2564 was resolved, can you check it again?

**Comment by niner** [comment]
The fix for #2564 was specific to CStruct, so I'd be surprised if there's a change in 99-misc.t

**Comment by AlexDaniel** [comment]
@dogbert17 have you tried reproducing this? It's not a big endian issue, so should probably also happen on amd64.

**Comment by AlexDaniel** [comment]
Yeah, I've been running it in a loop and got nothing. @robertlemmen can you perhaps check it again? Maybe something else fixed it…

**Comment by JJ** [comment]
Here too. It's run already many times. Different multipliers in test 3, but
not a problem.

**Comment by robertlemmen** [comment]
I just re-ran the original looped test on ppc64el, but with newer moar (g34fac5f4e), nqp (gbd7bf00b5) and rakudo (g5b231e099), which relatively quickly resulted in a:
```
===SORRY!===
Cannot invoke this object (REPR: Null; VMNull)
```

**Comment by AlexDaniel** [comment]
Was there any change in behavior during the last month?

**Comment by AlexDaniel** [comment]
I think this comment applies here too: https://github.com/rakudo/rakudo/issues/2576#issuecomment-465572758
Can't tell if this issue is completely new or if we just didn't notice it before for some reason, but for now we will have to let it be.

**Comment by dod38fr** [comment]
I now face a [similar issue on mipsel with rakudo 2019.11 ](https://buildd.debian.org/status/fetch.php?pkg=rakudo&arch=mipsel&ver=2019.11-3&stamp=1578442672&raw=0).

`make test` fails with:
```
t/08-performance/99-misc.t ...................................... 
1..4
ok 1 - eqaddr optimization for cmp exists
ok 2 - is 1999999000000 == 1999999000000
not ok 3 - was native .sum 2.2318754 at least 10x as fast as 19.2604672 (8.629723256145953x)
# Failed test 'was native .sum 2.2318754 at least 10x as fast as 19.2604672 (8.629723256145953x)'
# at t/08-performance/99-misc.t line 29
```
From the build log above, we can see that speed difference is between 8.6 and 8.9. I not really worried about that number on mipsel arch, so I propose to patch this test to allow successful build on mipsel (and this will also help build on armhf which tends to fail with a similar issue).

To help understand flapping tests, the tests are run after build with:
```
make test || make test || make test || make test || MVM_SPESH_DISABLE=1 make test
```

On mipsel, the last run (`MVM_SPESH_DISABLE=1 make test`) failed with:
```
t/04-nativecall/20-concurrent.t ................................. 
1..3
An operation first awaited:
  in block <unit> at t/04-nativecall/20-concurrent.t line 21

Died with the exception:
    Calling fib(Int) will never work with declared signature (int32 --> int32)
      in method arity-error at /<<PKGBUILDDIR>>/lib/NativeCall.pm6 (NativeCall) line 558
      in block fib at /<<PKGBUILDDIR>>/lib/NativeCall.pm6 (NativeCall) line 581
      in code  at t/04-nativecall/20-concurrent.t line 21
```

HTH

**Comment by niner** [comment]
The build script for the openSUSE package for rakudo simply deletes t/08-performance/99-misc.t before running `make test` because of these bogus failures:
https://build.opensuse.org/package/view_file/devel:languages:perl6/rakudo/rakudo.spec?expand=1

At this point I argue that the test does more harm than good. I'm not aware of any regression it ever cought and it creates lots of work for packagers and developers diagnosing problems that don't exist.

**Comment by niner** [comment]
@dod38fr the 04-nativecall/20-concurrent.t failure was fixed in https://github.com/rakudo/rakudo/commit/5c65a126c453f3f5e2b36e03261618a583e35586

**Comment by AlexDaniel** [comment]
Yeah, performance tests can be very annoying sometimes :(

**Comment by niner** [comment]
Even worse: the failing test in 08-performance/99-misc.t doesn't make sense at all. We're testing that X must be at least 10 times as fast as Y. So what if we make Y faster through spectacular optimizations. The test would tell us, that we broke rakudo by making it better!

And in this specific case, it's not even purely theoretical that this could happen. With escape analysis and quite a bit more magic, we may be able to get rid of the Int temporaries, thereby eliminating a performance difference.

**Comment by AlexDaniel** [comment]
> Even worse: the failing test in 08-performance/99-misc.t doesn't make sense at all. We're testing that X must be at least 10 times as fast as Y. So what if we make Y faster through spectacular optimizations. The test would tell us, that we broke rakudo by making it better!

Not really. It's a rakudo test, not a spectest. It's simply there as a canary.

**Comment by dod38fr** [comment]
> Yeah, performance tests can be very annoying sometimes :(

As this impacts all downstream distributions (Debian, OpenSuse...), I wonder if they should be run by default.

**Comment by niner** [comment]
> As this impacts all downstream distributions (Debian, OpenSuse...), I wonder
> if they should be run by default.

I argue that they should not. We should make them opt-in rather than opt-out. 
I'd be happy to put an environment assignment into the .bashrc on my dev 
workstation to spare packagers and users the trouble and the confusion about 
whether something is really broken or not.

**Comment by lizmat** [comment]
is this still an issue?

**Comment by dod38fr** [comment]
I've stepped down from packaging Raku for Debian. You'll have to ask Timo Paulssen for a follow up.
